src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/UseValidPlatformString.cs

```diff
@@ -21,6 +21,7 @@ public sealed class UseValidPlatformString : DiagnosticAnalyzer
         private static readonly ImmutableArray<string> methodNames = ImmutableArray.Create("IsOSPlatform", "IsOSPlatformVersionAtLeast");
         private const string IsPrefix = "Is";
         private const string VersionSuffix = "VersionAtLeast";
+        private const string macOS = nameof(macOS);
```

Comment:
Is there a place somewhere in the roslyn-analyzers repo where we have all the platform names predefined? Seems strange to have only this one defined here.

Also, wouldn't it be better to use a `const string` instead of a `nameof` that points to its own name? This pattern is a bit strange.

Reply:
> Is there a place somewhere in the roslyn-analyzers repo where we have all the platform names predefined? Seems strange to have only this one defined here.

Do not know, anyway for the analyzer the known names loaded from MSBuild known platforms list

> Also, wouldn't it be better to use a const string instead of a nameof that points to its own name? This pattern is a bit strange.

It can, but i wouldn't say that would be better. In this case, as you can see the variable name is the same as the string literal (no need to be different or no need to be more descriptive), in fact, it is not just a random string literal, it illustrates that it is a known OS platform name from MSBuild list used with the exact same value and casing from the list.